Add subctl show connections cmd

maayanf24 opened the pull request:
Resolves issue #469

Signed-off-by: Maayan Friedman <maafried@redhat.com>

tpantelis commented on pkg/subctl/cmd/show_connections.go:
```diff
@@ -0,0 +1,100 @@
+}
+
+func newConnectionsStatusFrom(gateway string, cluster string, remoteIp string, cableDriver string, subnets string, status submv1.ConnectionStatus) connectionsStatus {
+	v := connectionsStatus{
+		gateway:     gateway,
+		cluster:     cluster,
+		remoteIp:    remoteIp,
+		cableDriver: cableDriver,
+		subnets:     subnets,
+		status:      status,
+	}
+	return v
```
```suggestion
	return connectionsStatus{
		gateway:     gateway,
		cluster:     cluster,
		remoteIp:    remoteIp,
		cableDriver: cableDriver,
		subnets:     subnets,
		status:      status,
	}
```

tpantelis commented on pkg/subctl/cmd/show_connections.go:
```diff
@@ -0,0 +1,100 @@
+
+var showConnectionsCmd = &cobra.Command{
+	Use:   "connections",
+	Short: "Get information on your connections related to submariner",
```
```suggestion
	Short: "Show connections with other clusters",
```

tpantelis commented on pkg/subctl/cmd/show_connections.go:
```diff
@@ -0,0 +1,100 @@
+var showConnectionsCmd = &cobra.Command{
+	Use:   "connections",
+	Short: "Get information on your connections related to submariner",
+	Long:  `This command shows the status of submariner connections in your cluster.`,
```
```suggestion
	Long:  `This command shows information about submariner endpoint connections with other cluster.`,
```

tpantelis commented on pkg/subctl/cmd/show_connections.go:
```diff
@@ -0,0 +1,100 @@
+
+	gateways := existingCfg.Status.Gateways
+	if gateways == nil {
+		exitWithErrorMsg("no gateways found")
```
```suggestion
		exitWithErrorMsg("No endpoints found")
```

tpantelis commented on pkg/subctl/cmd/show_connections.go:
```diff
@@ -0,0 +1,100 @@
+func printConnections(connections []connectionsStatus) {
+	template := "%-20s%-16s%-16s%-24s%-40s%-16s\n"
+
+	fmt.Printf(template, "GATEWAY", "CLUSTER", "REMOTE_IP", "CABLE-DRIVER", "SUBNETS", "STATUS")
```
```suggestion
	fmt.Printf(template, "GATEWAY", "CLUSTER", "REMOTE IP", "CABLE DRIVER", "SUBNETS", "STATUS")
```

tpantelis commented on pkg/subctl/cmd/show_connections.go:
```diff
@@ -0,0 +1,100 @@
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type connectionsStatus struct {
```
```suggestion
type connectionStatus struct {
```

tpantelis commented:
@maayanf24 When addressing review comments, we generally commit a new patch rather than amending as it helps reviewers to see what actually changed. Unless it's a small patch and/or the changes were small.

maayanf24 commented:
> @maayanf24 When addressing review comments, we generally commit a new patch rather than amending as it helps reviewers to see what actually changed. Unless it's a small patch and/or the changes were small.

@tpantelis  Thanks for clarifying this

tpantelis commented on pkg/subctl/cmd/show_connections.go:
```diff
@@ -0,0 +1,99 @@
+var showConnectionsCmd = &cobra.Command{
+	Use:   "connections",
+	Short: "Show cluster connectivity information",
+	Long:  `This command shows information about submariner endpoint connections with other cluster.`,
```
```suggestion
	Long:  `This command shows information about submariner endpoint connections with other clusters.`,
```

mangelajo commented on pkg/subctl/cmd/show_connections.go:
```diff
@@ -0,0 +1,99 @@
+	status      submv1.ConnectionStatus
+}
+
+func newConnectionsStatusFrom(gateway string, cluster string, remoteIp string, cableDriver string, subnets string, status submv1.ConnectionStatus) connectionStatus {
```
Same comment as in the other patch, it's cleaner to create the structure right away, unless you had some business logic for creating the structure.

mangelajo commented on pkg/subctl/cmd/show_connections.go:
```diff
@@ -0,0 +1,99 @@
+	showCmd.AddCommand(showConnectionsCmd)
+}
+
+func getConnectionsStatus(status []connectionStatus) []connectionStatus {
```
```suggestion
func getConnectionsStatus() []connectionStatus {
	var status []connectionStatus
```

mangelajo commented on pkg/subctl/cmd/show_connections.go:
```diff
@@ -0,0 +1,99 @@
+}
+
+func showConnections(cmd *cobra.Command, args []string) {
+	var status []connectionStatus
+	status = getConnectionsStatus(status)
```
```suggestion
	status = getConnectionsStatus()
```
